### DISCUSSION TITLE
How to turn off detections of some objects ?

### DISCUSSION THREAD

**Discussion by jardaoravec** [ultralytics/yolov5#2463] [General]
Can I ask one more question about turning off detections for some of objects ? I noticed this Yolo is detecting really everything, things like keyboards or cups is it possible to turn some of them off somewhere in code ?

**Comment by glenn-jocher** [comment]
@jardaoravec yes you can opt-in to the classes you want to detect by passing a classes list:
```bash
python detect.py --weights yolov5s.pt --classes 0 2  # detect only persons and cars
```

or with YOLOv5 PyTorch Hub models:
```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')
model.classes = [0, 2]  # detect only persons and cars

# Images
dir = 'https://github.com/ultralytics/yolov5/raw/master/data/images/'
imgs = [dir + f for f in ('zidane.jpg', 'bus.jpg')]  # batch of images

# Inference
results = model(imgs)
results.print()  # or .show(), .save()
```

**Reply by jardaoravec** [reply]
@glenn-jocher  thank you ! Please can you give me list of names of classes ? I mean which number is what object. Thank you

**Reply by glenn-jocher** [reply]
Each dataset yaml file contains a list of class names for that dataset:

For COCO see data/coco.yaml.

**Reply by PiotrGebczyk** [reply]
and how to turn off the display of one object, e.g. a book

**Reply by glenn-jocher** [reply]
@PiotrGebczyk to detect only certain classes:
```
python detect.py --weights yolov5s.pt --classes 0 2  # detect only persons and cars
```

**Comment by Abduqayyum** [comment]
@glenn-jocher Hi. How to disable YOLOv5 model from detecting certain objects, if i trained my own model using yolov5. Is it possible to do so?
